test/unit-test/test_lib/test_analysis/test_channel.py

```diff
@@ -2,26 +2,85 @@
 from lib.analysis import channel
 import lib.util as util
 import os
+from mock import patch
 
-current_directory = os.path.dirname(os.path.realpath(__file__))
 
 class ChannelTest(unittest.TestCase):
 
     def setUp(self):
-        self.log_data = util.load_from_disk(current_directory+ "/data/log_data")
-        self.nicks = util.load_from_disk(current_directory+ "/data/nicks")
-        self.nick_same_list = util.load_from_disk(current_directory+ "/data/nick_same_list")
+        current_directory = os.path.dirname(os.path.abspath(__file__))
+        self.test_data_dir = current_directory + "/../../../data/test_lib/test_analysis/"
+        self.log_data = util.load_from_disk(self.test_data_dir+ "/channel/log_data")
+        self.nicks = util.load_from_disk(self.test_data_dir+ "/channel/nicks")
+        self.nick_same_list = util.load_from_disk(self.test_data_dir+ "/channel/nick_same_list")
+        self.to_graph = util.load_from_disk(self.test_data_dir + "/channel/to_graph")
+        self.connected_nick_list = util.load_from_disk(self.test_data_dir + "/channel/connected_nick_list")
 
     def tearDown(self):
         self.log_data = None
         self.nicks = None
         self.nick_same_list = None
 
-    def test_conv_len_conv_refr_time(self):
+    def mock_create_connected_nick_list(self,conn_comp_list):
+        for i in range(len(conn_comp_list)):
+            conn_comp_list[i] = list(conn_comp_list[i])
+        return conn_comp_list
+
+    def mock_get_nick_sen_rec(self, iter_range, nick_to_search, conn_comp_list, nick_sen_rec):
+        for i in xrange(iter_range):
+            if (i < len(conn_comp_list)) and (nick_to_search in conn_comp_list[i]):
+                nick_sen_rec = conn_comp_list[i][0]
+                break
+        return nick_sen_rec
+
+    def mock_correctLastCharCR(self,inText):
+        if len(inText) > 1 and inText[len(inText) - 1] == '\\':
+            inText = inText[:-1] + 'CR'
+        return inText
+
+    def mock_correct_last_char_list(self,rec_list):
+        for i in xrange(len(rec_list)):
+            if rec_list[i]:
+                rec_list[i] = self.mock_correctLastCharCR(rec_list[i])
+        return rec_list
+
+    def mock_rec_list_splice(self,rec_list):
```

Comment:
The above is true for all the mocks introduced. I don't understand how they are mocking the util function. @bhuvangupta008 Can you please explain, am I missing something?

Reply:
Essentially, we are lifting up the dependency of the channel module from the util module. The functions for which it needed the util modules are defined here.

Reply:
@bhuvangupta008 We should follow the same pattern used in other unit-tests. It would be better if you cache these outputs and read it in the test rather than writing a side-effect and using the same code.